### ISSUE TITLE
package.json is not used to resolve module path

### ISSUE THREAD

**Issue by givankin** [karma-runner/karma-commonjs#16]
If one has a dependency that looks like this:

```
node_modules/a-library/
  package.json
  main-library-file.js
```

And in `package.json`:

```
...
main: "./main-library-file"
...
```

Then there is a 

```
Error: Could not find module 'node_modules/a-library.js' from 'x.js'
```

From a brief look at source code I suppose that `karma-commonjs` doesn't parse `package.json`s. Maybe it's fine, doing it in the browser would be strange, but this means (correct me if I'm wrong) that I have to use _built_ JS to run tests. And that's what I'd like to avoid because build procedure can grow complex and slow and running it on every "Shift+F10" makes people sad.
What if a special "map" is introduced in config file, so we can do

```
'a-library': 'node-modules/a-library/main-library-file.js'
```

or

```
'node-modules/a-library/main-library-file.js:a-library'
```

somewhere, in the spirit of `browserify-shim` and similar libraries? This would also make `karma-commonjs` feasible in cases where the dependencies are not in `node_modules` folder (e.g. if `bower`+`debowerify` is used).

**Comment by abbakym** [comment]
Unfortunately can't get this to work in `0.0.13`.

I get an
````
Uncaught Error: Could not find module 'lodash'.
````

The error happens in the `loadAsDirectory` function: it checks if `lodash`'s `package.json` is in "existing files", and because it is not, tries to load `index.js` - after which everything fails. As "existing files" is effectively `window.__cjs_module__`, I assumed that I have to load the `package.json` explicitly.

But after I load it like this:
````
    files: [
      'node_modules/lodash/package.json',
      //...
    ],
    preprocessors: {
      'node_modules/lodash/package.json': ['commonjs'],
      //...
    },
````

It throws that `window.__cjs_module__` is undefined. Indeed, if I look at preprocessed `package.json`, it looks like this:

````
window.__cjs_module__["/<...>/node_modules/lodash/package.json"] = {...}
````

Whereas other preprocessed JS files all look like this:
````
window.__cjs_modules_root__ = "<...>/node_modules";window.__cjs_module__ = window.__cjs_module__ || {};window.__cjs_module__["<...>/foo.js"] = function() { ... }
````

Have I missed something? How is this supposed to be configured?

Thanks a lot in advance for your help.

**Comment by abbakym** [comment]
I finally got `package.json` loaded by placing it _after_ some common-js-preprocessed JS files in the `files` array in `karma.conf.js`. It loads fine without an exception, as `__cjs_module__` is non-empty when it loads.

However this line is now making trouble:
````
requiringPathEls.shift(); //cut of initial part coming from /
````

I suppose it is usable on Unix/Linux (though I can't clearly imagine why), but on Windows it strips the "`C:`" part. And later, when `loadAsDirectory` is matching the path to `package.json` to "existing files", it doesn't find one, because in "existing files" the path indeed _does have_ the "`C:`" part.

**Comment by DaveEmmerson** [comment]
@givankin did you ever resolve this? It's a year on and I'm going round in circles hitting the same issues and also got as far as stepping through the code to see what was going on and came to the same conclusion about the / stripping.

**Comment by jamesshore** [comment]
@DaveEmmerson The Windows issue is a bug, which I fixed in #54. Just waiting for @pkozlowski-opensource to merge it.

**Comment by givankin** [comment]
@DaveEmmerson what I ended up doing is creating a (pretty complicated) set of grunt tasks that assemble all the dependencies for me and put into a special folder specifically for tests, and then I have

```
commonjsPreprocessor: {
  modulesRoot: 'src/dependencies/'
},
```

in my `karma.conf.js` file. This is cumbersome, but once configured it generally continues to work.
